lib/reek/smells/attribute.rb

```diff
@@ -12,10 +12,18 @@ module Smells
     # +attr_reader+, +attr_writer+ and +attr_accessor+ -- including those
     # that are private.
     #
-    # TODO: Eliminate private attributes
     # TODO: Catch attributes declared "by hand"
     #
     class Attribute < SmellDetector
+      ATTR_DEFN_METHODS = [:attr, :attr_reader, :attr_writer, :attr_accessor]
+
+      def initialize(*args)
+        @visiblity_tracker = {}
+        @visiblity_mode = :public
+        @result = Set.new
+        super(args)
```

Comment:
We try to follow Liskov substitution principle, so we prefer a bare `super` where possible (as `super`with explicit arguments suggests there might be a difference in constructor signatures of the current class and its superclass, which, in turn, might suggest a Liskov substitution principle violation).